Subject: Avoid messages when quiet
From: Robert Schwebel <r.schwebel@pengutronix.de>

I've tried to use busybox with CONFIG_FEATURE_EXTRA_QUIET, but init's
startup banner does still appear. I don't understand the intended logic
behind the sniplet patched below: it is in an 'else' part that is only
executed if the syslog feature is not enabled; why should something
printed out differently if syslog is not enabled?

So this patch removes the duplicate full_write and delegates it to the
normal one.

The second part of the patch changes the file descriptor to stdout
instead of stderr. Init doesn't print out errors.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

ported to busybox-1.13.2

Signed-ogg-by: Marc Kleine-Budde <mkl@pengutronix.de>

---
 init/init.c |    6 ++++++
 1 file changed, 6 insertions(+)

Index: busybox-1.13.2/init/init.c
===================================================================
--- busybox-1.13.2.orig/init/init.c
+++ busybox-1.13.2/init/init.c
@@ -152,16 +152,22 @@ static void message(int where, const cha
 				}
 			}
 		}
+#if 0
 		if (where & L_LOG) {
 			full_write(log_fd, msg, l);
 			if (log_fd == STDERR_FILENO)
 				return; /* don't print dup messages */
 		}
+#endif
 	}
 
 	if (where & L_CONSOLE) {
 		/* Send console messages to console so people will see them. */
+#if 0
 		full_write(STDERR_FILENO, msg, l);
+#else
+		full_write(STDOUT_FILENO, msg, l);
+#endif
 	}
 }
 
